javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/OpenTelemetryInstaller.java

```diff
@@ -53,181 +31,60 @@ public void beforeByteBuddyAgent() {
     } else {
       log.info("Tracing is disabled.");
     }
   }
 
-  private static synchronized void installExporters(List<String> exporters, Properties config) {
-    for (String exporterName : exporters) {
-      SpanExporterFactory spanExporterFactory = findSpanExporterFactory(exporterName);
-      if (spanExporterFactory != null) {
-        installExporter(spanExporterFactory, config);
-      } else {
-        log.warn("No {} span exporter found", exporterName);
-      }
-
-      MetricExporterFactory metricExporterFactory = findMetricExporterFactory(exporterName);
-      if (metricExporterFactory != null) {
-        installExporter(metricExporterFactory, config);
-      } else {
-        log.debug("No {} metric exporter found", exporterName);
-      }
-
-      MetricServer metricServer = findMetricServer(exporterName);
-      if (metricServer != null) {
-        installMetricServer(metricServer, config);
-      } else {
-        log.debug("No {} metric server found", exporterName);
-      }
-    }
-  }
-
-  private static MetricExporterFactory findMetricExporterFactory(String exporterName) {
-    ServiceLoader<MetricExporterFactory> serviceLoader =
-        ServiceLoader.load(
-            MetricExporterFactory.class, OpenTelemetryInstaller.class.getClassLoader());
-
-    for (MetricExporterFactory metricExporterFactory : serviceLoader) {
-      if (metricExporterFactory.getNames().contains(exporterName)) {
-        return metricExporterFactory;
-      }
-    }
-    return null;
-  }
-
-  private static MetricServer findMetricServer(String exporterName) {
-    ServiceLoader<MetricServer> serviceLoader =
-        ServiceLoader.load(MetricServer.class, OpenTelemetryInstaller.class.getClassLoader());
-
-    for (MetricServer metricServer : serviceLoader) {
-      if (metricServer.getNames().contains(exporterName)) {
-        return metricServer;
-      }
-    }
-    return null;
-  }
-
-  private static SpanExporterFactory findSpanExporterFactory(String exporterName) {
-    ServiceLoader<SpanExporterFactory> serviceLoader =
-        ServiceLoader.load(
-            SpanExporterFactory.class, OpenTelemetryInstaller.class.getClassLoader());
-
-    for (SpanExporterFactory spanExporterFactory : serviceLoader) {
-      if (spanExporterFactory.getNames().contains(exporterName)) {
-        return spanExporterFactory;
-      }
-    }
-    return null;
-  }
-
-  private static synchronized void installExportersFromJar(String exporterJar, Properties config) {
-    URL url;
-    try {
-      url = new File(exporterJar).toURI().toURL();
-    } catch (MalformedURLException e) {
-      log.warn("Filename could not be parsed: " + exporterJar + ". Exporter is not installed");
-      log.warn("No valid exporter found. Tracing will run but spans are dropped");
-      return;
+  // OpenTelemetrySdkAutoConfiguration currently only supports configuration from environment. We
```

Comment:
But `io/opentelemetry/sdk/autoconfigure/ConfigProperties.java:24` does read from system properties?

And I still don't quite understand what does this method do :)

Reply:
By environment, I meant both env vars and system properties here.

This method is to forward properties to the autoconfiguration for agent's unique sources, file and `PropertySource`. We may or may not want to move the handling of these to autoconfigure, or do something different but this is to play with a first iteration and works ok I think.